arviz/data/inference_data.py

```diff
@@ -1394,10 +1394,11 @@ def extend(self, other, join="left"):
                 )
             dataset = getattr(other, group)
             setattr(self, group, dataset)
-            if group.startswith(WARMUP_TAG):
-                self._groups_warmup.append(group)
-            else:
-                self._groups.append(group)
+            if not hasattr(self, group):
```

Comment:
The current condition will always evaluate to false, the attribute is set in the previous line. We should check for it's presence in the `_groups` or `_groups_warmup` list with something like:

```
if group not in self._groups:
```
in which case it's probably easier to combine with the already present ifs